lib/rouge/lexers/mason.rb

```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+module Rouge
+  module Lexers
+    class Mason < TemplateLexer
+      title 'Mason'
+      desc 'HTML::Mason'
+      tag 'mason'
+      filenames '*.mi', '*.mc', '*.mas', '*.m', '*.mhtml', '*.mcomp', 'autohandler', 'dhandler'
+      mimetypes 'text/x-mason', 'application/x-mason'
+
+      def initialize(*)
+        super
+        @perl = Perl.new
+      end
+  
+      def self.detect?(text)
+        return false if text.doctype?(/[html|xml]/)
+        return true if text.doctype?
+      end
+
+      textblocks = %w(text doc)
+      perlblocks = %w(args flags attr init once shared perl cleanup filter)
+      components = %w(def method)
+  
+      state :root do
+        mixin :mason_tags
+      end
+
+      state :mason_tags do
+        rule /\s+/, Text::Whitespace
+
+        rule /<%(#{textblocks.join('|')})>/i, Keyword::Constant, :text_block

```

Comment:
How do ERB and/or other template lexers highlight the template tags? I just wonder if `Name::Tag` wouldn't be more appropriate (but maybe not since it wouldn't stand out as clear from the HTML tags).

Reply:
Yeah, we should probably make those other lexers consistently use `Comment::Preproc`.

Reply:
I just realized that if I use `Comment::Preproc`, this `<&|` is shown as `<&/` because they are highlighted with italics :(

Reply:
The token used doesn't by itself affect the output. It's the particular stylesheet that changes how tokens are displayed. If the correct token is `Comment::Preproc` then that's the token that should be used.

Reply:
And for what it's worth, `Comment::Preproc` definitely looks to be the correct token to use.

Reply:
The problem is that there are 3 themes that show `Comment::Preproc` with italics (Magritte, Thankful eyes, Tulip). I don't want to risk people seeing `|` as `/` because they might think the code is wrong...

Reply:
I'll have a bit more of a think but I'm pretty strongly of the opinion that that's a cause to update those themes: it's not a cause to misuse a token. The semantic meaning of the tokens should be the basis for their selection, not the way they'll look in our default themes.

Reply:
I still really feel like a better solution would be to change the way that the stylesheets highlight preprocessor comments (eg. instead of using italics, set a different background colour for these tokens) but I admit that that's a discussion that's better had separately, possibly in #1237. 

I had a look at the way Pygments does this and it uses its equivalent of the `Name::Tag` token (so does Chroma). I realise that this makes it difficult to differentiate between the HTML tags and the Mason tags but I think it's better to be consistent with Pygments while we determine how best to deal with this.

Reply:
Can we fix the stylesheet first? I don't feel comfortable making this change and then never having the stylesheet change merged. It will leave this bug that I definitely don't want to have.

Reply:
PR #1264 has been submitted to remove the use of italics in `Comment::Preproc` tokens in the three themes where they're currently in use.

Reply:
PR #1264 has now been merged.